reana_client/cli/quotas.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2017, 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA client quotas related commands."""
+
+import logging
+import sys
+
+import click
+from reana_client.cli.utils import (
+    add_access_token_options,
+    check_connection,
+    NotRequiredIf,
+)
+
+
+@click.group(help="Quota commands")
+def configuration_group():
+    """Quota commands."""
+    pass
+
+
+@click.option(
+    "--resource",
+    "resource",
+    cls=NotRequiredIf,
+    not_required_if="resources",
+    help="Specify quota resource. e.g. cpu, disk.",
+)
+@click.option(
+    "--resources",
+    "resources",
+    is_flag=True,
+    cls=NotRequiredIf,
+    not_required_if="resource",
+    help="Print available resources",
+)
+@click.option(
+    "--report",
+    "report",
+    type=click.Choice(["limit", "usage"], case_sensitive=False),
+    help="Specify quota report type. e.g. limit, usage.",
+)
+@configuration_group.command("quota-show")
+@click.pass_context
+@add_access_token_options
+@check_connection
+def quota_show(ctx, access_token, resource, resources, report):  # noqa: D301
+    """Show user quota.
+
+    The `quota-show"` command displays quota usage for the user.
+
+    Examples: \n
+    \t $ reana-client quota-show --resource disk --report limit\n
+    \t $ reana-client quota-show --resource disk --report usage\n
+    \t $ reana-client quota-show --resource disk\n
+    \t $ reana-client quota-show --resources
+    """
+    from reana_client.api.client import get_user_quotas
+
+    logging.debug("command: {}".format(ctx.command_path.replace(" ", ".")))
+
+    for p in ctx.params:
+        logging.debug("{param}: {value}".format(param=p, value=ctx.params[p]))
+
+    try:
+        quotas = get_user_quotas(access_token)
+
+        if resources:
+            return click.echo("\n".join(quotas.keys()))
+
+        if resource not in quotas.keys():
+            click.echo(
+                click.style(
+                    "Error: resource '{}' is not valid.\nAvailable resources are: '{}'.".format(
+                        resource, "', '".join(sorted(quotas.keys())),
+                    ),
+                    fg="red",
+                ),
+                err=True,
+            )
+            sys.exit(1)
+
+        # TODO: implement percentage calculation
+        if not report:
+            usage = quotas[resource]["usage"]
+            limit = quotas[resource]["limit"]
+            return click.echo("{} out of {} used".format(usage, limit))
```

Comment:
Idea: we can colour the message in accordance with the usage: green until 60%, yellow 60%-85%, red 85%-100%, WDYT? This can come later though.

Reply:
Nice idea, I left a `todo` comment for this to be implemented after we will have db models